pkg/api/image/formatter.go

```diff
@@ -60,18 +62,60 @@ func (h UploadActionHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request
 	rw.WriteHeader(http.StatusOK)
 }
 
-func (h UploadActionHandler) do(rw http.ResponseWriter, req *http.Request) error {
+func (h ImageHandler) do(rw http.ResponseWriter, req *http.Request) error {
 	vars := mux.Vars(req)
-	action := vars["action"]
-	switch action {
-	case actionUpload:
+	operation := ""
+	if _, found := vars["action"]; found {
+		operation = vars["action"]
+	} else if _, found := vars["link"]; found {
+		operation = vars["link"]
+	}
+	switch operation {
+	case operationUpload:
 		return h.uploadImage(rw, req)
+	case operationDownload:
+		return h.downloadImage(rw, req)
 	default:
 		return apierror.NewAPIError(validation.InvalidAction, "Unsupported action")
 	}
 }
 
-func (h UploadActionHandler) uploadImage(rw http.ResponseWriter, req *http.Request) error {
+func (h ImageHandler) downloadImage(rw http.ResponseWriter, req *http.Request) error {
+	vars := mux.Vars(req)
+	namespace := vars["namespace"]
+	name := vars["name"]
+
+	bkimgName := fmt.Sprintf("%s-%s", namespace, name)
```

Comment:
should check both namespace and name are not "" ?

Reply:
I am thinking about that both `namespace` and `name` come from API. That should not be empty or there is something error that would have happened early.

Reply:
As @FrankYang0529 suggested, I change the output file name to the user-readable image name.
It would get the Image first, so it will fail when getting the image failure.
Maybe we do not need another check?

Reply:
Please take a look of LH API, hardcode LH backend may break someday. Both upload and download.

```
vendor/github.com/longhorn/longhorn-manager/types/types.go
	DefaultAPIPort           = 9500
func GetDefaultManagerURL() string {
	return "http://longhorn-backend:" + strconv.Itoa(DefaultAPIPort) + "/v1"
}
```

Reply:
making them constant would be good enough at this stage: https://github.com/harvester/harvester/blob/538d41fe793c41360cb170f128206c768ce83735/pkg/util/constants.go#L22
We can do whole code base refactoring by using the [namespace value pass via Helm chart values](https://github.com/harvester/harvester/blob/538d41fe793c41360cb170f128206c768ce83735/deploy/charts/harvester/values.yaml#L377) in the future.

Reply:
Add into constants, and use the related type on both upload and download. 
Please check it again, thanks. :)